server/main_chatbot.py

```
@@ -648,9 +699,32 @@ async def handle_connection(websocket) -> None:
+        logger.info(f"WebSocket authenticated via token: user={token_user_id}")
+
+        # Eagerly warm caches for token-authenticated users
+        if token_user_id:
+            asyncio.get_event_loop().call_soon(
+                lambda uid=token_user_id: asyncio.ensure_future(update_api_cost_cache_async(uid))
+            )
+            try:
+                from chat.backend.agent.tools.mcp_preloader import preload_user_tools, update_user_activity
+                preload_user_tools(token_user_id)
+                update_user_activity(token_user_id)
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)